Tweak: Add `max-height` control to containers and sections [ED-9575]

m1ga opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* adding `max-height` value to the container element

## Description
An explanation of what is done in this PR

* adding `max-height` value to the container element

## Test instructions
This PR can be tested by following these steps:

* create a container
* add a value the the new `max-height` property (below min-height)
* switch overflow mode to `auto` or `hidden`

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #



https://user-images.githubusercontent.com/4334997/212478029-e39967bb-0a89-4a47-915c-5f8bb66a4d81.mp4

m1ga committed:
feat: container max-height

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/elementor/elementor?pullRequest=21013) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/elementor/elementor?pullRequest=21013) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/elementor/elementor?pullRequest=21013) it.</sub>

rami-elementor commented on includes/elements/container.php:
```diff
@@ -443,6 +443,28 @@ protected function register_container_layout_controls() {
+						'min' => 0,
+						'max' => 1440,
+					],
+					'vh' => [
+						'min' => 0,
+						'max' => 100,
+					],
```
Can be removed as this is has already defined in `includes\controls\base-units.php`

m1ga replied:
I just copied it from existing `'Min Height'`. The values are there too :smile: 
I'll remove them :+1:

rami-elementor commented on includes/elements/container.php:
```diff
@@ -443,6 +443,28 @@ protected function register_container_layout_controls() {
+				'range' => [
+					'px' => [
+						'min' => 0,
+						'max' => 1440,
```
Why limiting the `max-height` to `1440`? What if I need a larger container?

m1ga replied:
Shall I use a high value or is there an unlimited option?

m1ga replied:
I've raised the value to 10000. Otherwise the slide is very hard to control. User can always enter a higher value if needed.

rami-elementor commented:
For consistency, you should add the `max-height` control to sections too.

m1ga committed:
remove vh range

m1ga committed:
higher slider value, add max-height to section

rami-elementor commented on includes/elements/container.php:
```diff
@@ -443,6 +439,24 @@ protected function register_container_layout_controls() {
+			[
+				'label' => esc_html__( 'Max Height', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', 'vh', 'custom' ],
```
```suggestion
				'size_units' => [ 'px', 'vh', 'vw', 'custom' ],
```

rami-elementor replied:
I've noticed that sections use three unit types (`px`, `vh`, `vw`) while containers use two (`px`, `vh`). Let's fix that too.

rodolphebertozzo replied:
Is it really relevant to add the vw for the height, it seems to me that this unit is only suitable for the width?

m1ga replied:
I don't mind having it there. If it's not there there will be one user that will ask for it :smile:

rodolphebertozzo replied:
Obviously, this is not disturbing in itself, other builders have all the units everywhere to offer something complete, because the code structure is different. Elementor wants to offer more units, just as the base code requires units to be added to containers and elements on a case by case basis for each of them, so we might as well think about this detail.

m1ga committed:
Update includes/elements/container.php

Co-authored-by: Rami Yushuvaev <92088692+rami-elementor@users.noreply.github.com>

m1ga committed:
add vw

matipojo committed:
Internal: Lighthouse workflow fails [HTS-432] (#22028)

maork-elementor committed:
Internal: Add target_commitish - softprops/action-gh-release (Core) [ED-10422] (#22064)

TzviRabinovitch committed:
Internal: Fix version in new action (#22069)

TzviRabinovitch committed:
Internal: Fix version in new action (#22069)

maork-elementor committed:
Internal: Update zip file name (#22070)

maork-elementor committed:
Internal: Update zip file name (#22070)

maork-elementor committed:
Internal: Fix zip GitHub release (#22072)

maork-elementor committed:
Internal: Fix zip GitHub release (#22072)

arielk committed:
Internal: Facelift fixes (for patch version) (#22086)

ArmanPurtoian committed:
Fix: Horizontal scrolling [ED-10302] (#22097)

arielk committed:
Fix: Panel responsive button clickable (#22073) [ED-10466] (#22104)

arielk committed:
Internal: Fix font size in the new facelift (#22109)

Nevoss committed:
Internal: Admin can change DB attributes from "Replace URL" - this option should be blocked [ED-9500] (#22129)

Co-authored-by: maork-elementor <maork@elementor.com>

udidol committed:
Internal: Changelog for v3.12.2 (#22160)

elementorbot committed:
Bump 3.12.2

elementorbot committed:
Merge develop into master

m1ga committed:
Merge branch 'master' into containerMaxHeight

m1ga commented:
any news if this can be included?

maork-elementor committed:
Internal: Marge 3.13 into master [ED-10728] (#22446)

TzviRabinovitch committed:
Internal: Changelog v3.13.4 (#22577)

maork-elementor committed:
Fix: Arbitrary File Write via Archive Extraction (Zip Slip) [ED-10576] (#22585)

maork-elementor committed:
Fix: Templates responsive setting won't update unless 'Regenerate CSS' is used [ED-10594] (#22588)

StyleShit committed:
Fix: Save as default styling issues when adding a widget with a click (#22587) [ED-10896] (#22607)

StyleShit committed:
Internal: Editor V2 - Change document name automatically [ED-9330] (#22616)

KingYes committed:
New: AI Images (#22579)

Co-authored-by: rotemee <61740845+rotemee@users.noreply.github.com>
Co-authored-by: Ohad <ohad@elementor.com>
Co-authored-by: Rotem Ben-Itzhak <rotem@elementor.com>
Co-authored-by: Ariel Klikstein <ariel@elementor.com>

rotemee committed:
Tweak: AI Images UI CP [ED-10886] (#22628)

KingYes committed:
Tweak: Images AI Tweaks [ED-10815] (#22665)

## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:
https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no
spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

---------

Co-authored-by: Raz Ohad <ohad@elementor.com>
Co-authored-by: Ariel Klikstein <ariel@elementor.com>
Co-authored-by: Dmytro Maziar <dmytrom@elementor.com>

Commit:
Fix: New page created from the recently edited dropdown should created as a page [ED-10958] (#22698)

maork-elementor committed:
Fix: Only the 1st loop grid widget will load [ED-10945] (#22662) (#22702)

KingYes committed:
Tweak: Improve errros on AI [ED-10881] (#22699) (#22734)

rotemee committed:
New: Image AI Tools CP [ED-11009] (#22741)

Co-authored-by: Ohad <ohad@elementor.com>
Co-authored-by: Yakir Sitbon <kingyes1@gmail.com>

KingYes committed:
Tweak: Remove Image AI hidden experiment [ED-10988] (#22695) (#22742)

hein-obox committed:
Internal: [Core] [NC] [A11Y] Nested Carousel Accessibility - Beta Branch [ED-10852] (#22738)

rotemee committed:
Internal: UI Version Update [ED-11038] (#22748)

hein-obox committed:
Fix: Video background [...] in Tabs widget - beta [ED-10932] (#22752)

rami-elementor committed:
Fix: Dark mode background color for select controls in the Editor [ED-10748] (#22761)

rami-elementor committed:
Tweak: Use "media_types" array in Media controls [ED-11052] (#22764)

KingYes committed:
Tweak: Remove AI button from non-image media control [ED-11030] (#22759) (#22762)

Co-authored-by: Ariel Klikstein <ariel@elementor.com>

murzin-nikita committed:
Internal: Fix flaky PW tests beta [ED-10774]  (#22754)

arielk committed:
Internal: Announcement popup with images promotion 3.14 [ED-10822] (#22766)

## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:
https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no
spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

rotemee committed:
Fix: Image AI Tools CP [ED-11053] (#22768)

KingYes committed:
Tweak: Arrange AI button when has responsive switchers [ED-11057] (#22770)

TzviRabinovitch committed:
Revert "Fix: Video background  .. [ED-11102] " (#22807)

TzviRabinovitch committed:
Internal: Changelog for v3.14.0 (#22810)

KingYes committed:
Internal: Remove plugins promotion [ED-11094] (#22839)

rotemee committed:
Fix: The default image should not be considered CP [ED-11152] (#22845)

KingYes committed:
Tweak: Image AI Update error message for filtered words [ED-11093]

louiswol94 committed:
Fix: Video background added [...] only works on the first tab in Tabs widget [ED-11104] (#22879)

bainternet committed:
Internal: Image AI Add reconnect button to images modal - CP [ED-11047] (#22875)

Co-authored-by: Yakir Sitbon <kingyes1@gmail.com>

bainternet committed:
Internal: Image Prompt library downloads checkpoint - CP [ED-10818] (#22880)

Co-authored-by: Ariel Klikstein <ariel@elementor.com>

bainternet committed:
Internal: Images AI code tweaks [ED-11045] CP (#22889)

bainternet committed:
Tweak: Change upgrade alert from info to error [ED-11158] - CP (#22881)

Co-authored-by: Ariel Klikstein <ariel@elementor.com>

Nevoss committed:
Internal: Generic warning message for upload JSON [ED-11138] (#22898)

TzviRabinovitch committed:
Internal: Changelog v3.14.1 (#22900)

murzin-nikita committed:
Tweak: Unify apperance of stretch button to main [ED-11073] (#22909)

murzin-nikita committed:
Tweak: Unify appearance of horizontal stretch button to main [ED-11073] (#22912)

murzin-nikita committed:
Fix: Stretch option is not working as expected in Tabs widget 3.14 [ED-11073] (#22901)

maork-elementor committed:
Fix: Create RC tag - Remove sync to master on GA [ED-11229] (#22904) (#22921)

Nevoss committed:
Internal: Add JSON warning message in apps [ED-11255] (#22956)

m1ga committed:
Merge branch 'master' into containerMaxHeight

StyleShit committed:
Tweak: Improved Ajax permissions functionality for better security enforcement [ED-11233] (#23038)

TzviRabinovitch committed:
Revert "Merge remote-tracking branch 'Elementor/3.14'"

This reverts commit 8e285b38d3285d17411411193e781956aea37f19.

TzviRabinovitch committed:
Merge remote-tracking branch 'Elementor/3.14'

TzviRabinovitch committed:
Internal: Workaround to allow screenshooter tests for 3.16

m1ga committed:
Merge branch 'master' into containerMaxHeight

m1ga commented:
Still no feedback here?

davseve committed:
Update elementor.php (#23482)

m1ga committed:
Merge branch 'master' into containerMaxHeight

m1ga commented:
@yotamselementor I think it would be easier to redo the PR as there are so many merge conflicts now.
The only question is: is there any interest in this PR at all since I've made that over a year ago and nothing happened.
